packages/falcon-ui/src/components/NumberInput.tsx

```diff
@@ -79,18 +81,31 @@ class NumberInputInnerDOM extends React.Component<NumberInputInnerDOMProps> {
   };
 
   render() {
-    const { className, invalid, ...remaining } = this.props;
+    const { className, invalid, min, max, increaseText, decreaseText, value, ...remaining } = this.props;
     const { themableProps, rest } = extractThemableProps(remaining);
+    const currentValue = this.inputRef.current ? this.inputRef.current.value : value;
 
     return (
       <Box {...themableProps} className={className}>
-        <button type="button" aria-hidden onClick={this.stepDown} className="-inner-input-step-down-element">
+        <button
+          type="button"
+          onClick={this.stepDown}
+          aria-label={decreaseText}
+          className="-inner-input-step-down-element"
+          disabled={currentValue <= min}
+        >
           <Icon src="numberInputDown" fallback="−" />
         </button>
 
-        <input ref={this.inputRef} min={5} type="number" {...rest} />
+        <input ref={this.inputRef} min={min} value={value} type="number" {...rest} />
```

Comment:
```suggestion
        <input ref={this.inputRef} min={min} max={max} value={value} type="number" {...rest} />
```
I think `max` wasn't properly set because it was destructured from props.